### ISSUE TITLE
DivideError: integer division error in ClimaAtmos

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimaCore.jl#1265]
## Describe the bug
Here is the [build](https://buildkite.com/clima/climaatmos-ci/builds/9886#01885044-6496-430a-a61f-bed49999e4ac).

The stack trace until it gets unwieldy is:

```julia
┌ Error: ClimaAtmos simulation crashed. Stacktrace for failed simulation
│   exception =
│    DivideError: integer division error
│    Stacktrace:
│      [1] div
│        @ ./int.jl:288 [inlined]
│      [2] div
│        @ ./div.jl:281 [inlined]
│      [3] div
│        @ ./div.jl:336 [inlined]
│      [4] cld
│        @ ./div.jl:293 [inlined]
│      [5] _configure_threadblock
│        @ /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/ClimaCore/A0lxf/src/Spaces/dss_cuda.jl:6 [inlined]
│      [6] dss_local!(#unused#::ClimaComms.CUDADevice, perimeter_data::ClimaCore.DataLayouts.VIFH{NamedTuple{(), Tuple{}}, 12, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, perimeter::ClimaCore.Spaces.Perimeter2D{4}, topology::ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float32}, ClimaCore.Meshes.NormalizedBilinearMap}, Vector{CartesianIndex{3}}, Array{Int64, 3}, CUDA.CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}})
│        @ ClimaCore.Spaces /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/ClimaCore/A0lxf/src/Spaces/dss_cuda.jl:97
│      [7] weighted_dss_internal!(data::ClimaCore.DataLayouts.VIJFH{NamedTuple{(), Tuple{}}, 4, CUDA.CuArray{Float32, 5, CUDA.Mem.DeviceBuffer}}, space::ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Spaces.CellCenter, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float32}, ClimaCore.Meshes.NormalizedBilinearMap}, Vector{CartesianIndex{3}}, Array{Int64, 3}, CUDA.CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{4}, ClimaCore.Geometry.SphericalGlobalGeometry{Float32}, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.LatLongPoint{Float32}, Float32, StaticArraysCore.SMatrix{2, 2, Float32, 4}}, 4, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float32, 4, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float32, ClimaCore.Geometry.UVVector{Float32}}, 4, CUDA.CuArray{Float32, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, ClimaCore.Topologies.IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float32}, Tuple{Symbol, Symbol}}, Vector{ClimaCore.Geometry.ZPoint{Float32}}}, NamedTuple{(:bottom, :top), Tuple{Int64, Int64}}}, ClimaCore.Spaces.Flat, ClimaCore.Geometry.SphericalGlobalGeometry{Float32}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.LatLongZPoint{Float32}, Float32, StaticArraysCore.SMatrix{3, 3, Float32, 9}}, 4, CUDA.CuArray{Float32, 5, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.LatLongZPoint{Float32}, Float32, StaticArraysCore.SMatrix{3, 3, Float32, 9}}, 4, CUDA.CuArray{Float32, 5, CUDA.Mem.DeviceBuffer}}}, hspace::ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float32}, ClimaCore.Meshes.NormalizedBilinearMap}, Vector{CartesianIndex{3}}, Array{Int64, 3}, CUDA.CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.Mem.DeviceBuffer}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, NamedTuple{(), Tuple{}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.Mem.DeviceBuffer}}, ClimaCore.Spaces.Quadratures.GLL{4}, ClimaCore.Geometry.SphericalGlobalGeometry{Float32}, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.LatLongPoint{Float32}, Float32, StaticArraysCore.SMatrix{2, 2, Float32, 4}}, 4, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IJFH{Float32, 4, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float32, ClimaCore.Geometry.UVVector{Float32}}, 4, CUDA.CuArray{Float32, 3, CUDA.Mem.DeviceBuffer}}, NamedTuple{(), Tuple{}}}, dss_buffer::ClimaCore.Spaces.DSSBuffer{ClimaComms.SingletonGraphContext, ClimaCore.DataLayouts.VIFH{NamedTuple{(), Tuple{}}, 12, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}}, CUDA.CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Int64, 1, CUDA.Mem.DeviceBuffer}})
│        @ ClimaCore.Spaces /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/ClimaCore/A0lxf/src/Spaces/dss.jl:356
│      [8] weighted_dss_internal!
│        @ /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/ClimaCore/A0lxf/src/Spaces/dss.jl:325 [inlined]
│      [9] weighted_dss_internal!
│        @ /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/ClimaCore/A0lxf/src/Fields/Fields.jl:369 [inlined]
│     [10] macro expansion
│        @ /central/scratch/esm/slurm-buildkite/climaatmos-ci/9886/climaatmos-ci/src/tendencies/hyperdiffusion.jl:161 [inlined]
│     [11] macro expansion
│        @ /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/cpu/packages/NVTX/d9htq/src/macro.jl:119 [inlined]
│     [12] tracer_hyperdiffusion_tendency!(Yₜ::ClimaCore.Fields.FieldVector{Float32, NamedTuple{(:c, :f), Tuple{Field{(:ρ, :uₕ, :ρe_tot)} (trunc 
```

## To Reproduce
Here is the PR where it's happening: https://github.com/CliMA/ClimaAtmos.jl/pull/1706

@sriharshakandala, can you take a look at this asap?

**Comment by simonbyrne** [comment]
Issue appears to be that `nfid = 0`.

**Comment by charleskawczynski** [comment]
Or should we fix the DSS?

**Comment by charleskawczynski** [comment]
Yes, see https://github.com/CliMA/ClimaAtmos.jl/pull/1714

Can we still throw a more informative error in dss?

**Comment by sriharshakandala** [comment]
That said, I believe exception handling should happen much earlier.

**Comment by simonbyrne** [comment]
there's no reason we can't support empty fields, can't we just make DSS a no-op in this case?

**Comment by dennisYatunin** [comment]
@charleskawczynski If we need to avoid DSS-ing empty fields in ClimaAtmos, we could add a simple check to the beginning of `tracer_hyperdiffusion_tendency!`, maybe something like
```
isempty(propertynames(ᶜ∇²specific_tracers)) && return nothing
```

**Comment by charleskawczynski** [comment]
Closed by #1270.
